vermouth/graph_utils.py

```diff
@@ -194,51 +129,127 @@ def rate_match(residue, bead, match):
                for rdx, bdx in match.items())
 
 
-def make_residue_graph(mol):
+def _items_with_common_values(graph, nodes=None):
+    """
+    Finds all node attributes of nodes in graph that all have the same values.
+    Returns a dict of node attribute/common value pairs
+
+    Parameters
+    ----------
+    graph: networkx.Graph
+    nodes: collections.abc.Iterable or None
+        If None, consider all nodes. All nodes must be in graph.
+
+    Returns
+    -------
+    dict
+    """
+    if not nodes:
+        nodes = set(graph.nodes)
+    common_attrs = defaultdict(list)
+    for idx in nodes:
+        for key, val in graph.nodes[idx].items():
+            common_attrs[key].append(val)
+    common_attrs = dict(common_attrs)
```

Comment:
Nice one. That simplifies matters significantly.
The cast from defaultdict to dict is needed since defaultdict will never raise a KeyError for missing elements. It would create them instead. Maybe not super relevant in this case, but sure to hide serious bugs in others.